# Análisis de la información presupuestal presentada en Transparencia Presupuestaria

En el sitio http://www.transparenciapresupuestaria.gob.mx/es/PTP/datos_presupuestarios_abiertos se puede encontrar un csv que contiene datos presupuestarios desde el 2008 hasta el primer trimestre del 2017.
Nos interesa conocer cómo podemos cruzar la información de nuestra bbdd de contrataciones públicas con información presupuestaria.

In [1]:
import pandas as pd

In [2]:
pre = pd.read_csv('Descargas/presupuesto_mexico_2008_2017.csv')

/home/felix/miniconda2/lib/python2.7/site-packages/IPython/core/interactiveshell.py:2718: DtypeWarning: Columns (1,3,26,27,38) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [3]:
pre.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2315726 entries, 0 to 2315725
Data columns (total 39 columns):
CICLO                      int64
ID_RAMO                    object
DESC_RAMO                  object
ID_UR                      object
DESC_UR                    object
GPO_FUNCIONAL              int64
DESC_GPO_FUNCIONAL         object
ID_FUNCION                 int64
DESC_FUNCION               object
ID_SUBFUNCION              int64
DESC_SUBFUNCION            object
ID_AI                      int64
DESC_AI                    object
ID_MODALIDAD               object
DESC_MODALIDAD             object
ID_PP                      int64
DESC_PP                    object
ID_CONCEPTO                int64
DESC_CONCEPTO              object
ID_TIPOGASTO               int64
DESC_TIPOGASTO             object
ID_FF                      int64
DESC_FF                    object
MONTO_APROBADO             float64
MONTO_EJERCIDO             float64
ID_ENTIDAD_FEDERATIVA      float64
ENTIDAD_F

Con esta información podemos conocer cuál ha sido el presupuesto gastado para el proyecto del nuevo aeropuerto para el año 2016.

Primero ubicamos el gasto para el año 2016

In [7]:
pre2016 = pre[pre.CICLO == 2016]

In [9]:
pre2016.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 276435 entries, 1811445 to 2087879
Data columns (total 39 columns):
CICLO                      276435 non-null int64
ID_RAMO                    276435 non-null object
DESC_RAMO                  276435 non-null object
ID_UR                      276435 non-null object
DESC_UR                    276435 non-null object
GPO_FUNCIONAL              276435 non-null int64
DESC_GPO_FUNCIONAL         276435 non-null object
ID_FUNCION                 276435 non-null int64
DESC_FUNCION               276435 non-null object
ID_SUBFUNCION              276435 non-null int64
DESC_SUBFUNCION            276435 non-null object
ID_AI                      276435 non-null int64
DESC_AI                    276435 non-null object
ID_MODALIDAD               276435 non-null object
DESC_MODALIDAD             276435 non-null object
ID_PP                      276435 non-null int64
DESC_PP                    276435 non-null object
ID_CONCEPTO                276435 non-

Podemos identificar que existen varias columnas relacionadas con el monto: "MONTO APROBADO", "MONTO EJERCIDO", "MONTO_EJERCICIO", "MONTO_MODIFICADO", "MONTO_DEVENGADO", "MONTO_PAGADO", "MONTO_ADEFAS".
Se puede descargar en la misma página donde se descarga el .csv el diccionario de las columnas del csv. 

El "MONTO_DEVENGADO" es aquel que representa la obligación de pago por la contratación de bienes y servicios por lo que usaremos este número. También conocemos el ID_CLAVE_CARTERA del Nuevo Aeropuerto (NAICM). Esta información la puedes encontrar en la página: https://www.gob.mx/shcp/acciones-y-programas/cartera-publica-de-inversion-16287

In [10]:
pre2016[pre2016.ID_CLAVE_CARTERA == "1409JZL0005"]

CICLO ID_RAMO                            DESC_RAMO ID_UR  \
1845369   2016       9         Comunicaciones y Transportes   KDH   
1845370   2016       9         Comunicaciones y Transportes   KDH   
1933582   2016      16  Medio Ambiente y Recursos Naturales   B00   
1933594   2016      16  Medio Ambiente y Recursos Naturales   B00   
1933595   2016      16  Medio Ambiente y Recursos Naturales   B00   
1933597   2016      16  Medio Ambiente y Recursos Naturales   B00   
1933599   2016      16  Medio Ambiente y Recursos Naturales   B00   
1933607   2016      16  Medio Ambiente y Recursos Naturales   B00   
1933613   2016      16  Medio Ambiente y Recursos Naturales   B00   

                                                   DESC_UR  GPO_FUNCIONAL  \
1845369  Grupo Aeroportuario de la Ciudad de México, S....              3   
1845370  Grupo Aeroportuario de la Ciudad de México, S....              3   
1933582                         Comisión Nacional del Agua              2   
1933594                         Comisión Nacional del Agua              2   
1933595                         Comisión Nacional del Agua              2   
1933597                         Comisión Nacional del Agua              2   
1933599                         Comisión Nacional del Agua              2   
1933607                         Comisión Nacional del Agua              2   
1933613                         Comisión Nacional del Agua              2   

           DESC_GPO_FUNCIONAL  ID_FUNCION          DESC_FUNCION  \
1845369  Desarrollo Económico           5            Transporte   
1845370  Desarrollo Económico           5            Transporte   
1933582     Desarrollo Social           1  Protección Ambiental   
1933594     Desarrollo Social           1  Protección Ambiental   
1933595     Desarrollo Social           1  Protección Ambiental   
1933597     Desarrollo Social           1  Protección Ambiental   
1933599     Desarrollo Social           1  Protección Ambiental   
1933607     Desarrollo Social           1  Protección Ambiental   
1933613     Desarrollo Social           1  Protección Ambiental   

         ID_SUBFUNCION                        ...                          \
1845369              4                        ...                           
1845370              4                        ...                           
1933582              3                        ...                           
1933594              3                        ...                           
1933595              3                        ...                           
1933597              3                        ...                           
1933599              3                        ...                           
1933607              3                        ...                           
1933613              3                        ...                           

        MONTO_MODIFICADO  MONTO_DEVENGADO  MONTO_PAGADO MONTO_ADEFAS  \
1845369     0.000000e+00     0.000000e+00  0.000000e+00         0.00   
1845370     0.000000e+00     0.000000e+00  0.000000e+00         0.00   
1933582     0.000000e+00     0.000000e+00  0.000000e+00         0.00   
1933594     2.214662e+06     2.214662e+06  2.214662e+06         0.00   
1933595     6.366062e+07     6.366062e+07  6.355145e+07    109171.83   
1933597     1.357314e+09     1.357314e+09  1.307124e+09  50190018.50   
1933599     1.721166e+08     1.721166e+08  1.717438e+08    372842.78   
1933607     1.374610e+08     1.374610e+08  1.277379e+08   9723041.24   
1933613     9.864079e+06     9.864079e+06  9.853736e+06     10343.91   

        ID_CAPITULO                            DESC_CAPITULO  \
1845369        6000                        Inversion publica   
1845370        6000                        Inversion publica   
1933582        3000                      Servicios generales   
1933594        3000                      Servicios generales   
1933595        5000  Bienes muebles, inmuebles e intangibles   


In [11]:
pre2016NAICM =pre2016[pre2016.ID_CLAVE_CARTERA == "1409JZL0005"]

In [13]:
pre2016NAICM.sort_values("MONTO_DEVENGADO", ascending=False).groupby("DESC_UR").MONTO_DEVENGADO.sum()

DESC_UR
Comisión Nacional del Agua                                  1.742631e+09
Grupo Aeroportuario de la Ciudad de México, S.A. de C.V.    0.000000e+00
Name: MONTO_DEVENGADO, dtype: float64

Vemos como la suma del monto devengado "0.0" para el Grupo Aeroportuario de la Ciudad de México, S.A. de C.V. (GACM) Ahora revisamos los partidas específicas que involucra sólo al GACM. Para esto creamos un data frame con sólo filas del GACM.

In [19]:
pre2016NAICM.head(9)

CICLO ID_RAMO                            DESC_RAMO ID_UR  \
1845369   2016       9         Comunicaciones y Transportes   KDH   
1845370   2016       9         Comunicaciones y Transportes   KDH   
1933582   2016      16  Medio Ambiente y Recursos Naturales   B00   
1933594   2016      16  Medio Ambiente y Recursos Naturales   B00   
1933595   2016      16  Medio Ambiente y Recursos Naturales   B00   
1933597   2016      16  Medio Ambiente y Recursos Naturales   B00   
1933599   2016      16  Medio Ambiente y Recursos Naturales   B00   
1933607   2016      16  Medio Ambiente y Recursos Naturales   B00   
1933613   2016      16  Medio Ambiente y Recursos Naturales   B00   

                                                   DESC_UR  GPO_FUNCIONAL  \
1845369  Grupo Aeroportuario de la Ciudad de México, S....              3   
1845370  Grupo Aeroportuario de la Ciudad de México, S....              3   
1933582                         Comisión Nacional del Agua              2   
1933594                         Comisión Nacional del Agua              2   
1933595                         Comisión Nacional del Agua              2   
1933597                         Comisión Nacional del Agua              2   
1933599                         Comisión Nacional del Agua              2   
1933607                         Comisión Nacional del Agua              2   
1933613                         Comisión Nacional del Agua              2   

           DESC_GPO_FUNCIONAL  ID_FUNCION          DESC_FUNCION  \
1845369  Desarrollo Económico           5            Transporte   
1845370  Desarrollo Económico           5            Transporte   
1933582     Desarrollo Social           1  Protección Ambiental   
1933594     Desarrollo Social           1  Protección Ambiental   
1933595     Desarrollo Social           1  Protección Ambiental   
1933597     Desarrollo Social           1  Protección Ambiental   
1933599     Desarrollo Social           1  Protección Ambiental   
1933607     Desarrollo Social           1  Protección Ambiental   
1933613     Desarrollo Social           1  Protección Ambiental   

         ID_SUBFUNCION                        ...                          \
1845369              4                        ...                           
1845370              4                        ...                           
1933582              3                        ...                           
1933594              3                        ...                           
1933595              3                        ...                           
1933597              3                        ...                           
1933599              3                        ...                           
1933607              3                        ...                           
1933613              3                        ...                           

        MONTO_MODIFICADO  MONTO_DEVENGADO  MONTO_PAGADO MONTO_ADEFAS  \
1845369     0.000000e+00     0.000000e+00  0.000000e+00         0.00   
1845370     0.000000e+00     0.000000e+00  0.000000e+00         0.00   
1933582     0.000000e+00     0.000000e+00  0.000000e+00         0.00   
1933594     2.214662e+06     2.214662e+06  2.214662e+06         0.00   
1933595     6.366062e+07     6.366062e+07  6.355145e+07    109171.83   
1933597     1.357314e+09     1.357314e+09  1.307124e+09  50190018.50   
1933599     1.721166e+08     1.721166e+08  1.717438e+08    372842.78   
1933607     1.374610e+08     1.374610e+08  1.277379e+08   9723041.24   
1933613     9.864079e+06     9.864079e+06  9.853736e+06     10343.91   

        ID_CAPITULO                            DESC_CAPITULO  \
1845369        6000                        Inversion publica   
1845370        6000                        Inversion publica   
1933582        3000                      Servicios generales   
1933594        3000                      Servicios generales   
1933595        5000  Bienes muebles, inmuebles e intangibles   


Podemos observar que ninguna de las columnas referente al proyecto del nuevo aeropuerto, donde participa el GACM, no tienen datos en ninguna de las columnas de "MONTO". Esto es un error en la base de datos y queda documentado aquí para decidir qué pasos tomar.

Ahora, teniendo como antecedente que no existen datos completos en el proyecto del NAICM. Búsquemos si la Unidad Responsable del nuevo aeropuerto tiene más gastos no relacionados con el NAICM.

In [20]:
pre2016GACM = pre2016[pre2016.DESC_UR == "Grupo Aeroportuario de la Ciudad de México, S.A. de C.V."]

In [21]:
pre2016GACM.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 102 entries, 1845271 to 2016497
Data columns (total 39 columns):
CICLO                      102 non-null int64
ID_RAMO                    102 non-null object
DESC_RAMO                  102 non-null object
ID_UR                      102 non-null object
DESC_UR                    102 non-null object
GPO_FUNCIONAL              102 non-null int64
DESC_GPO_FUNCIONAL         102 non-null object
ID_FUNCION                 102 non-null int64
DESC_FUNCION               102 non-null object
ID_SUBFUNCION              102 non-null int64
DESC_SUBFUNCION            102 non-null object
ID_AI                      102 non-null int64
DESC_AI                    102 non-null object
ID_MODALIDAD               102 non-null object
DESC_MODALIDAD             102 non-null object
ID_PP                      102 non-null int64
DESC_PP                    102 non-null object
ID_CONCEPTO                102 non-null int64
DESC_CONCEPTO              102 non-null object

In [18]:
pre2016GACM.ID_CLAVE_CARTERA.value_counts()

00000000000    100
1409JZL0005      2
Name: ID_CLAVE_CARTERA, dtype: int64

Aquí se identifica que el Grupo Aeroportuario no tiene catalagodo 100 de sus 102 gastos por lo que necesitamos otro identificador para conocer su monto gastado.

In [22]:
pre2016GACM.head(10)

CICLO ID_RAMO                     DESC_RAMO ID_UR  \
1845271   2016       9  Comunicaciones y Transportes   KDH   
1845272   2016       9  Comunicaciones y Transportes   KDH   
1845273   2016       9  Comunicaciones y Transportes   KDH   
1845274   2016       9  Comunicaciones y Transportes   KDH   
1845275   2016       9  Comunicaciones y Transportes   KDH   
1845276   2016       9  Comunicaciones y Transportes   KDH   
1845277   2016       9  Comunicaciones y Transportes   KDH   
1845278   2016       9  Comunicaciones y Transportes   KDH   
1845279   2016       9  Comunicaciones y Transportes   KDH   
1845280   2016       9  Comunicaciones y Transportes   KDH   

                                                   DESC_UR  GPO_FUNCIONAL  \
1845271  Grupo Aeroportuario de la Ciudad de México, S....              3   
1845272  Grupo Aeroportuario de la Ciudad de México, S....              3   
1845273  Grupo Aeroportuario de la Ciudad de México, S....              3   
1845274  Grupo Aeroportuario de la Ciudad de México, S....              3   
1845275  Grupo Aeroportuario de la Ciudad de México, S....              3   
1845276  Grupo Aeroportuario de la Ciudad de México, S....              3   
1845277  Grupo Aeroportuario de la Ciudad de México, S....              3   
1845278  Grupo Aeroportuario de la Ciudad de México, S....              3   
1845279  Grupo Aeroportuario de la Ciudad de México, S....              3   
1845280  Grupo Aeroportuario de la Ciudad de México, S....              3   

           DESC_GPO_FUNCIONAL  ID_FUNCION DESC_FUNCION  ID_SUBFUNCION  \
1845271  Desarrollo Económico           5   Transporte              4   
1845272  Desarrollo Económico           5   Transporte              4   
1845273  Desarrollo Económico           5   Transporte              4   
1845274  Desarrollo Económico           5   Transporte              4   
1845275  Desarrollo Económico           5   Transporte              4   
1845276  Desarrollo Económico           5   Transporte              4   
1845277  Desarrollo Económico           5   Transporte              4   
1845278  Desarrollo Económico           5   Transporte              4   
1845279  Desarrollo Económico           5   Transporte              4   
1845280  Desarrollo Económico           5   Transporte              4   

                               ...                         MONTO_MODIFICADO  \
1845271                        ...                               2142586.24   
1845272                        ...                                 56526.42   
1845273                        ...                               5843889.72   
1845274                        ...                               1327139.09   
1845275                        ...                                407465.50   
1845276                        ...                                162986.18   
1845277                        ...                                199015.27   
1845278                        ...                                159308.40   
1845279                        ...                                     0.00   
1845280                        ...                                914910.88   

         MONTO_DEVENGADO MONTO_PAGADO MONTO_ADEFAS ID_CAPITULO  \
1845271       2142586.24   2142586.24          0.0        1000   
1845272         56526.42     56526.42          0.0        1000   
1845273       5843889.72   5843889.72          0.0        1000   
1845274       1327139.09   1327139.09          0.0        1000   
1845275        407465.50    407465.50          0.0        1000   
1845276        162986.18    162986.18          0.0        1000   
1845277        199015.27    199015.27          0.0        1000   
1845278        159308.40    159308.40          0.0        1000   
1845279             0.00         0.00          0.0        1000   
1845280        914910.88    914910.88          0.0        1000   

                DESC_CAPITULO ID_PARTIDA_GENERICA  \
1845271  Servicios perso

In [25]:
pre2016GACM.ID_CAPITULO.value_counts()

3000    52
2000    30
1000    13
4000     5
6000     2
Name: ID_CAPITULO, dtype: int64